client/src/main/scala/org/http4s/client/oauth1/Header.scala

```diff
@@ -0,0 +1,77 @@
+package org.http4s.client.oauth1
+
+import cats.Show
+import cats.kernel.Order
+import cats.implicits._
+sealed trait Header {
+  val headerName: String
+  val headerValue: String
+  def toTuple: (String, String) =
+    headerName -> encode(headerValue)
+}
+
+object Header {
+
+  case class Consumer(override val headerValue: String, secret: String) extends Header {
+    override val headerName: String = "oauth_consumer_key"
+  }
+
+  case class Token(override val headerValue: String, secret: String) extends Header {
+    override val headerName: String = "oauth_token"
+  }
+
+  case class Realm(override val headerValue: String) extends Header {
+    override val headerName: String = "realm"
+  }
+
+  case class Custom(headerName: String, headerValue: String) extends Header
+
+  case class SignatureMethod(override val headerValue: String = "HMAC-SHA1") extends Header {
+    override val headerName: String = "oauth_signature_method"
+  }
+
+  case class Timestamp(
+      override val headerValue: String = (System.currentTimeMillis() / 1000).toString)
+      extends Header {
+    override val headerName: String = "oauth_timestamp"
+  }
+
+  case class Nonce(override val headerValue: String = System.nanoTime.toString) extends Header {
+    override val headerName: String = "oauth_nonce"
+  }
+
+  case class Version(override val headerValue: String = "1.0") extends Header {
+    override val headerName: String = "oauth_version"
+  }
+
+  case class Callback(override val headerValue: String) extends Header {
+    override val headerName: String = "oauth_callback"
+  }
+
+  case class Verifier(override val headerValue: String) extends Header {
+    override val headerName: String = "oauth_verifier"
+  }
+
+  implicit val sortHeaders: Order[Header] = new Order[Header] {
```

Comment:
Nit we've started following the cats convention of naming with the package, type name, and typeclass name, to avoid any accidental shadowing.  So, `http4sClientOauth1SortForHeaders`.